crypto/ec/ec_key.c

```diff
@@ -1112,19 +1134,67 @@ static int ecdsa_keygen_pairwise_test(EC_KEY *eckey, OSSL_CALLBACK *cb,
     OSSL_SELF_TEST_onbegin(st, OSSL_SELF_TEST_TYPE_PCT,
                            OSSL_SELF_TEST_DESC_PCT_ECDSA);
 
-    sig = ECDSA_do_sign(dgst, dgst_len, eckey);
-    if (sig == NULL)
+    group = EC_KEY_get0_group(eckey);
+    curve_name = OPENSSL_strdup(EC_curve_nid2nist(EC_GROUP_get_curve_name(group)));
+    keylen = EC_KEY_key2buf(eckey, POINT_CONVERSION_UNCOMPRESSED, &keyoct, bnctx);
+    if (keylen == 0)
+        goto err;
+    privlen = EC_KEY_priv2buf(eckey, &privoct);
+    if (privlen == 0)
+        goto err;
+    paramskey[0] = OSSL_PARAM_construct_utf8_string(OSSL_PKEY_PARAM_GROUP_NAME, curve_name, 0);
+    paramskey[1] = OSSL_PARAM_construct_octet_string(OSSL_PKEY_PARAM_PUB_KEY, keyoct, keylen);
+    paramskey[2] = OSSL_PARAM_construct_BN(OSSL_PKEY_PARAM_PRIV_KEY, privoct, privlen);
```

Comment:
This is not how BIGNUM parameters are created. Instead see , for example: https://github.com/openssl/openssl/blob/871182d29dc24bbb51ff7a4d952ff9d9868f14b5/crypto/evp/p_lib.c#L2385-L2392
You can allocate the buffer if you prefer that over asserting that some fixes size is sufficient.